packages/cli-kit/src/public/node/git.ts

```diff
@@ -63,6 +63,38 @@
   appendFileSync(filePath, fileContent)
 }
 
+/**
+ * Add an entry to an existing .gitignore file.
+ *
+ * If the .gitignore file doesn't exist, or if the entry is already present,
+ * no changes will be made.
+ *
+ * @param root - The directory containing the .gitignore file.
+ * @param entry - The entry to add to the .gitignore file.
+ */
+export function addToGitIgnore(root: string, entry: string): void {
+  const gitIgnorePath = joinPath(root, '.gitignore')
+
+  if (!fileExistsSync(gitIgnorePath)) {
+    // When the .gitignore file does not exist, the CLI should not be opinionated about creating it
+    return
+  }
+
+  const gitIgnoreContent = readFileSync(gitIgnorePath).toString()
+  const eol = detectEOL(gitIgnoreContent)
+
+  if (gitIgnoreContent.split(eol).some((line) => line.trim() === entry.trim())) {
+    // The file already existing in the .gitignore
+    return
+  }
+
+  if (gitIgnoreContent.endsWith(eol)) {
+    writeFileSync(gitIgnorePath, `${gitIgnoreContent}${entry}${eol}`)
+  } else {
+    writeFileSync(gitIgnorePath, `${gitIgnoreContent}${eol}${entry}${eol}`)
```

Comment:
Question! Have you seen the `appendFile` function from `fs`?
It basically appends your content to the end of a file, maybe it already handles all the EOL stuff?? 🤔